sdk/client.go

```diff
@@ -83,3 +85,19 @@ func (c *Client) getServiceStatus(resPath string) (*api.MetaMessage, error) {
 	}
 	return msg, nil
 }
+
+func (c *Client) postServiceAction(resPath string, req interface{}) error {
+	resp, resErr := c.postResource(resPath, req)
+	// Since the response body can have more information about the error, we try to parse it
+	if resErr != nil && resp == nil {
+		return fmt.Errorf("postResource: %w", resErr)
+	}
+	msg := api.MetaMessage{}
+
+	if err := json.Unmarshal(resp, &msg); err != nil {
+		// if the response is not a MetaMessage, it's probably a success message
+		// Therefore we just return the original error from the request
+		return resErr
+	}
+	return Error{Message: msg}
```

Comment:
It's currently not clear to me when it succeeds and when it fails.

Reply:
Well the HTTP request can succeed, but the requested action might fail. for example when user requests to start a packetloss on a non existing network interface. The HTTP request to the endpoint succeeds, but the underlying service returns an error in form of `MetaMessage`  with details.
So there are two layers of errors here.
Not sure how to make it clearer.

Reply:
The new comment explains it better now. Also, the new structure looks good :+1: